packages/pg-protocol/src/parser.ts

```diff
@@ -102,19 +113,94 @@ export class Parser {
 
       const fullMessageLength = CODE_LENGTH + length
 
-      if (fullMessageLength + offset <= combinedBuffer.byteLength) {
+      if (fullMessageLength + offset <= combinedBufferFullLength) {
         const message = this.handlePacket(offset + HEADER_LENGTH, code, length, combinedBuffer)
         callback(message)
         offset += fullMessageLength
       } else {
         break
       }
     }
+    this.consumeBuffer({
+      combinedBuffer,
+      combinedBufferOffset: offset,
+      combinedBufferLength,
+      reuseRemainingBuffer,
+      combinedBufferFullLength,
+    })
```

Comment:
I have a question for you....a long time ago it was more efficient to make "backing classes" for things you'd return frequently as v8 can optimize them much more aggressively.  You are using a type `CombinedBuffer` and constructing it w/ the object shorthand here & returning it w/ the object shorthand on line `172`.  Do you think it'd make a difference to actually make this a class?
something like this...
```
class CombinedBuffer {
  constructor(public combinedBuffer: buffer = combinedBuffer, public combinedBufferOffset: number = offset, public combinedBufferLength: number = length, public reuseRemainingBuffer: boolean = reuse, public combinedBufferFullLength: number = fullLength) {}
}
```
and then return it with `this.consumeBuffer(new CombinedBuffer(combinedBuffer, offset, combinedBufferLength, reuseRemainingBuffer, combinedBufferFullLength))`

? 

Staying on top of how v8 optimizes things is a moving target so I'm not sure if you have any experience around the perf diff of using dynamically created objects versus proper "classes."  I wouldn't normally sweat it but this is a white code code path & it might be a small perf win here.

Reply:
You might also be able to allocate a single instance of `CombinedBuffer` on the parser class itself & just set the properties on it versus allocating new ones every time.   Again, a micro-optimization but those things can add up when the code is called so frequently. 

I can play around w/ this too & push to your branch (I think github lets me do that?) if you're cool w/ it & think it's an okay idea.

Reply:
That is an interesting solution, why don't you let me research it (as I'm not too familiar with v8 bits) and try to enhance it